taichi/python/export_lang.cpp

```diff
@@ -82,6 +82,20 @@ void export_lang(py::module &m) {
 #undef PER_EXTENSION
       .export_values();
 
+  py::class_<DataType>(m, "DataType")
+      .def(py::self == py::self)
+      .def(py::pickle(
+          [](const DataType &dt) { return py::make_tuple((std::size_t)dt); },
```

Comment:
nit: Since `dt` represents a type, when I first saw this, I got confused by thinking this is taking the size of the type. I'd suggest that we have an explicit function here, rather than doing `operator size_t()`. That way it is also easier to code search... :)

Reply:
nice idea. It would be better to have a function which does the exact same thing(or just call `operator size_t()`)

Reply:
>  It would be better to have a function
Yeah +1. I'd suggest to be explicit and not to have `operator size_t()` at all. This kind of implicitness could very likely go in an unintended manner. (Similarly, we try to add `explicit` at constructors that take in a single parameter). Maybe https://stackoverflow.com/a/22164767/12003165 and http://ptgmedia.pearsoncmg.com/imprint_downloads/informit/aw/meyerscddemo/DEMO/MEC/MC2_FR.HTM could be some good arguments on this?

Reply:
Oh I see. Thanks!